pip/req/req_install.py

```diff
@@ -343,11 +343,18 @@ def setup_py(self):
         assert self.source_dir, "No source dir for %s" % self
         try:
             import setuptools  # noqa
-        except ImportError:
+        except ImportError as e:
+            # Python 2 and 3 variants:
+            if e.args[0] in (
+                    "No module named setuptools",
+                    "No module named 'setuptools'"):
+                suggestion = "Please install setuptools."
+            else:
+                suggestion = "Check your setuptools installation: %s" % e
```

Comment:
Using the traceback module to format the error would be better than % formatting the exception - or perhaps log the details to pip's log and point people at that?